lib/alloc.h

```diff
@@ -20,6 +20,58 @@
+#define XCALLOC(n, type)       ((type *) xcalloc(n, sizeof(type)))
+#define MALLOCARRAY(n, type)   ((type *) mallocarray(n, sizeof(type)))
+#define XMALLOCARRAY(n, type)  ((type *) xmallocarray(n, sizeof(type)))
+
+#define ALLOCA(type)           ALLOCARRAY(1, type)
+#define MALLOC(type)           MALLOCARRAY(1, type)
+#define XMALLOC(type)          XMALLOCARRAY(1, type)
+#define REALLOC(ptr, type)     REALLOCARRAY(ptr, 1, type)
+#define REALLOCF(ptr, type)    REALLOCARRAYF(ptr, 1, type)
+
+#define REALLOCARRAY(ptr, n, type)                                            \
+({                                                                            \
+	__auto_type  p_ = (ptr);                                              \
+                                                                              \
+	static_assert(__builtin_types_compatible_p(typeof(p_), type *), "");  \
+                                                                              \
+	(type *) reallocarray(p_, n, sizeof(type));                           \
+})
+
+#define REALLOCARRAYF(ptr, n, type)                                           \
+({                                                                            \
+	__auto_type  p_ = (ptr);                                              \
+                                                                              \
+	static_assert(__builtin_types_compatible_p(typeof(p_), type *), "");  \
+                                                                              \
+	(type *) reallocarrayf(p_, n, sizeof(type));                          \
+})
+
+#define XREALLOCARRAY(ptr, n, type)                                           \
+({                                                                            \
+	__auto_type  p_ = (ptr);                                              \
+                                                                              \
+	static_assert(__builtin_types_compatible_p(typeof(p_), type *), "");  \
+                                                                              \
+	(type *) xreallocarray(p_, n, sizeof(type));                          \
+})
+
+#define ALLOCARRAY(n, type)                                                   \
+({                                                                            \
+	type    *p_;                                                          \
+	size_t  n_ = (n);                                                     \
+                                                                              \
+	if (n_ != 0 && n_ > SSIZE_MAX / n_)                                   \
+		p_ = NULL;                                                    \
+	else                                                                  \
+		p_ = alloca(sizeof(type) * n_);                               \
+                                                                              \
+	p_;                                                                   \
```

Comment:
If you wanted to align the `\` characters at the end of the line you missed your goal.

Reply:
I align them to the 79 column on purpose :)

I learnt that from NGINX, where they do that.  The reason is that in a 80 column terminal you see that there's nothing to the right of the escape character.  If you put `\` in column 80, then you could accidentally have some character in column 81, and you might not see it, because it's after the right margin.  So if you put it in column 79, you still have 1 column empty, which is the proof that the escape indeed escapes the newline, and not some hidden character.